src/commands/bans/banned.ts

```diff
@@ -0,0 +1,28 @@
+import {Command} from "@ubccpsc310/bot-base";
+import {Client, Message} from "discord.js";
+import {getBannedWord} from "../../controllers/BanController";
+
+const banned: Command = {
+    name: "banned",
+    description: "Lists the banned word (if any)",
+    usage: "banned",
+    procedure: async (client: Client, message: Message) => {
+        let bannedWord = await getBannedWord();
+        const middle = (bannedWord.length / 2.0) + 0.5; // Adding 0.5 is the equivalent of ensuring it always rounds up
+        // Find the middle vowel and replace it with a *
+        for (let pos = 0; pos < middle; pos++) {
+            if (middle + pos < bannedWord.length && /[aeiou]/.test(middle[middle + pos])) {
+                // Replace character with a *
+                bannedWord = bannedWord.substring(0,middle + pos) + "*" + bannedWord.substring(middle + pos+1);
+                break;
+            } else if (middle - pos < bannedWord.length && /[aeiou]/.test(middle[middle - pos])) {
+                bannedWord = bannedWord.substring(0,middle - pos) + "*" + bannedWord.substring(middle - pos+1);
+                break;
```

Comment:
Just a suggestion, I don't think you need the explicit `break` here.

You're using an `if-else`, which ensures that only one of the branches will be executed (unless I read the  conditionals wrong and it's somehow possible for both of them to be executed).

Reply:
I think the break is to end the loop. As in only replace one vowel

Reply:
Yea, we only want to replace one vowel which is why there is a break. If I had used continue (or whatever the ts equivalent is), then it wouldn't be needed. Because of the break, It doesn't need to be an else, they could be separate ifs. I think it is just a bit easier to understand with the else but I'm happy to change that.